bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/nano/WlanThermoNanoV1Handler.java

```diff
@@ -0,0 +1,229 @@
+            try {
+                String json = gson.toJson(c);
+                URI uri = config.getUri("/setchannels");
+                int status = httpClient.POST(uri).content(new StringContentProvider(json), "application/json")
+                        .timeout(5, TimeUnit.SECONDS).send().getStatus();
+                if (status == 401) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "No or wrong login credentials provided. Please configure username/password for write access to WlanThermo!");
+                    break;
+                } else if (status != 200) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Failed to update channel "
+                            + c.getName() + " on device, Statuscode " + status + " on URI " + uri.toString());
+                    logger.debug("Payload sent: {}", json);
+                } else {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (TimeoutException | ExecutionException | URISyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Failed to update channel " + c.getName() + " on device: " + e.getMessage());
+            } catch (InterruptedException e) {
+                logger.debug("Push interrupted. {}", e.getMessage());
+                return;
+            }
+        }
+
+        // push update for pitmaster channels
+        try {
+            String json = gson.toJson(data.getPitmaster().getPm());
+            URI uri = config.getUri("/setpitmaster");
+            int status = httpClient.POST(uri).content(new StringContentProvider(json), "application/json")
+                    .timeout(5, TimeUnit.SECONDS).send().getStatus();
+            if (status == 401) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "No or wrong login credentials provided. Please configure username/password for write access to WlanThermo!");
+            } else if (status != 200) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Failed to update pitmaster channel on device, Statuscode " + status + " on URI "
+                                + uri.toString());
+                logger.debug("Payload sent: {}", json);
+            } else {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (TimeoutException | ExecutionException | URISyntaxException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Failed to update pitmaster channel on device: " + e.getMessage());
+        } catch (InterruptedException e) {
+            logger.debug("Push interrupted. {}", e.getMessage());
+        }
```

Comment:
This code is nearly identical to the code right above it. Please consider refactoring this into a separate method.

Reply:
Done